controllers/rcj_soccer_referee_supervisor/referee/referee.py

```diff
@@ -1,44 +1,248 @@
+
+    def _add_initial_position_noise(
+        self, translation: List[float]
+    ) -> List[float]:
+        """Return translation with noise added to x and z coordinates.
+
+        Args:
+            translation (list): x, y and z coordinates
+
+        Returns:
+            list: new x, y, and z coordinates
+        """
+        level = self.initial_position_noise
+        return [
+            translation[0] + (random.random() - 0.5) * level,
+            translation[1],
+            translation[2] + (random.random() - 0.5) * level,
+        ]
+
+    def add_event_subscriber(self, subscriber: EventHandler):
+        """Add new event subscriber.
+
+        Args:
+            subscriber (EventHandler): Instance inheriting EventHandler
+        """
+        self.eventer.subscribe(subscriber)
+
+    def add_event_message_to_queue(self, message: str):
+        """Add new message to the message queue.
+
+        Args:
+            message (str): Message to be added to the queue.
+        """
+        if len(self.event_messages_to_draw) >= MAX_EVENT_MESSAGES_IN_QUEUE:
+            self.event_messages_to_draw.pop(0)
+
+        self.event_messages_to_draw.append((self.time, message))
+
+    def process_and_draw_event_messages(self):
+        """Process and draw event messages from the queue"""
+        messages = []
+        for time, msg in self.event_messages_to_draw:
+            messages.append("{} - {}".format(time_to_string(time), msg))
+
+        self.sv.draw_event_messages(messages)
+
+    def reset_checkers(self, object_name: str):
+        """Reset rule checkers for the specified object.
+
+        Args:
+            object_name (str): Either "ball" or the robot's name.
+        """
+        self.progress_check[object_name].reset()
+        if object_name != "ball":
+            self.penalty_area_check[object_name].reset()
+
+    def reset_ball_position(self):
+        """Reset the position of the ball."""
+        self.sv.set_ball_position(BALL_INITIAL_TRANSLATION)
+        self.ball_stop = 2
+        self.reset_checkers("ball")
+
+    def reset_robot_position(self, robot_name: str):
+        """Reset robot's position to the initial one.
+
+        Args:
+            robot_name (str): The robot to reset the position for
+        """
+        self.sv.reset_robot_velocity(robot_name)
+
+        translation = ROBOT_INITIAL_TRANSLATION[robot_name].copy()
+        translation = self._add_initial_position_noise(translation)
+
+        self.sv.set_robot_position(robot_name, translation)
+        self.sv.set_robot_rotation(
+            robot_name, ROBOT_INITIAL_ROTATION[robot_name]
+        )
+
+        self.reset_checkers(robot_name)
+
+    def reset_positions(self):
+        """
+        Reset the positions of the ball as well as the robots to the initial
+        position.
+        """
+        self.reset_ball_position()
+
+        # reset the robot positions
+        for robot in ROBOT_NAMES:
+            self.reset_robot_position(robot)
 
```

Comment:
Well, this is indeed quite strange... After 201 lines Git/GitHub decided there is no difference here?

Reply:
Many methods were moved from supervisor to referee so that might be the reason.